### ISSUE TITLE
Can't test navigation between Flutter Web pages using Flutter Driver

### ISSUE THREAD

**Issue by mark-zs** [flutter/flutter#61111]
Flutter Driver has only very basic documentation for simple tests like finding a button and pressing the button. Anyone know if I can do other operations like navigate to a specific page from driver (e.g. using a # url fragment), test a link which leads to an external site, check visual setup of the page against an image, and other such tests which would be standard in Silenium and the like?  
Or alterantively a strategy for using Silenium/Ghost Inspector with Flutter Web

Thanks!

**Comment by darshankawar** [comment]
Hi @mark-zs,
I see there's a similar open issue, [25465](https://github.com/flutter/flutter/issues/25465)
Please follow up on that issue, I'm closing the current one as duplicate.
If you disagree, please write in the comments and I will reopen it.
Thank you
Related to

**Comment by mark-zs** [comment]
Hi @darshankawar .  I don't think that's a duplicate.  That was about interacting with external apps.  My main request is a very basic one.  That driver is able to navigate within the flutter app.  Not sure how I can use Driver to test app workflows without that. Thanks.

**Comment by darshankawar** [comment]
@mark-zs, fair enough. I thought the one I listed as original comes closer to your request.

**Comment by yjbanov** [comment]
`FlutterDriver` is meant to be cross-platform, so if we add an API for this, it would need to be something that also makes sense on mobile. So the feature might be to support pushing/popping routes. This wouldn't be a web-specific feature (so I'll remove the `platform-web` label from the issue).

As a workaround, consider using [requestData](https://api.flutter.dev/flutter/flutter_driver/FlutterDriver/requestData.html), which allows running arbitrary code on the app side at the driver test's request. For example, you can pass a URL and have the browser open it.
